cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in

```diff
@@ -188,8 +188,18 @@ else()
   set(copy_mpi_command "")
 endif()
 
+if(@BUILD_SHARED_LIBS@) #if using shared libs overide boost config
+  set(Boost_USE_STATIC_LIBS OFF)
+  if("@MSVC@")
+    set(BOOST_ALL_DYN_LINK ON)
```

Comment:
Clean this variable if we are outside of the `if` conditions. You can add `set(BOOST_ALL_DYN_LINK "")` before the first `if`.